code/game/objects/structures/lattice.dm

```diff
@@ -96,6 +96,34 @@
 	desc = "A heavily reinforced catwalk used to build bridges in hostile environments. It doesn't look like anything could make this budge."
 	resistance_flags = INDESTRUCTIBLE
 
+/obj/structure/lattice/lava
+	name = "heatproof support lattice"
+	desc = "A specialized support beam for building across lava. Watch your step."
+	icon = 'icons/obj/smooth_structures/catwalk.dmi'
+	icon_state = "catwalk-0"
+	base_icon_state = "catwalk"
+	number_of_rods = 2
+	color = "#5286b9ff"
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_LATTICE)
+	canSmoothWith = list(SMOOTH_GROUP_LATTICE, SMOOTH_GROUP_FLOOR, SMOOTH_GROUP_WALLS, SMOOTH_GROUP_TURF, SMOOTH_GROUP_WINDOW_FULLTILE)
+	resistance_flags = INDESTRUCTIBLE
+
+/obj/structure/lattice/lava/deconstruction_hints(mob/user)
+	to_chat(user, "<span class='notice'>The supporting rods look like they could be <b>cut</b>.</span>, but the <i>heat treatment will shatter off</i>.")
+
+/obj/structure/lattice/attackby(obj/item/C, mob/user, params)
```

Comment:
same as the other comment, why `C` and not `I` here for the `obj/item/` variable?